core/zms/src/main/rdl/Role.tdl

```diff
@@ -48,6 +48,7 @@ type RoleMeta Struct {
     String userAuthorityFilter (optional); //membership filtered based on user authority configured attributes
     String userAuthorityExpiration (optional); //expiration enforced by a user authority configured attribute
     Int32 groupExpiryDays (optional); //all groups in the domain roles will have specified max expiry days
+    Map<SimpleName,SimpleName> tags (optional); //key-value pair tags
```

Comment:
After thinking about this a little bit more, we should make the value as CompoundName. This would give us to have a little bit more flexibility to have .'s in our names - e.g. prod.us-west-2, etc.